packages/celotool/src/e2e-tests/utils.ts

```diff
@@ -497,35 +283,48 @@ export function getContext(gethConfig: GethTestConfig) {
         gethConfig.migrationOverrides
       )
     }
+
     await killGeth()
     await sleep(2)
+
     // Snapshot the datadir after the contract migrations so we can start from a "clean slate"
     // for every test.
     for (const instance of gethConfig.instances) {
-      await snapshotDatadir(instance)
+      await snapshotDatadir(gethConfig.runPath, instance, verbose)
     }
   }
 
   const restart = async () => {
     await killGeth()
+
+    if (gethConfig.useBootnode) {
+      await killBootnode()
```

Comment:
`killGeth` kills all running processes with `geth` in the name, so I don't think you should have to separately kill the bootnode. https://github.com/H34D/celo-monorepo/blob/a7e958900d46dc3396879cb91bf358959cd8f1cc/packages/celotool/src/e2e-tests/utils.ts#L64-L67

Reply:
bootnodes are started with the `bootnode` command not `geth` so killGeth will not kill bootnodes.